enterprise/server/remote_execution/dirtools/dirtools.go

```diff
@@ -357,6 +403,13 @@ func UploadTree(ctx context.Context, env environment.Env, dirHelper *DirHelper,
 				if err != nil {
 					return nil, err
 				}
+				symlinkNode, err := uploadSymlinkFn(fullPath, target, info)
+				if err != nil {
+					return nil, err
+				}
+
+				txInfo.FileCount += 1
+				txInfo.BytesTransferred += int64(len(symlinkNode.GetTarget()))
```

Comment:
If I'm understanding the documentation of `output_symlinks` correctly, it seems like all we need to do is for each `output_path`, if it's a symlink, then just add the symlink metadata (name, target) to `output_symlinks`. I would expect the logic here to just be something like

```
if dirHelper.IsOutputPath(fqfn) {
  actionResult.OutputSymlinks = append(actionResult.OutputSymlinks, &repb.SymlinkNode{Name: trimPathPrefix(fqfn, rootDir), Target: target})
}
```

maybe @iain-macdonald can double-check my understanding since I think he implemented output_paths originally

Reply:
https://app.buildbuddy.io/invocation/8e025ac8-fb37-4bcd-97c0-dc4001b64c32?actionDigest=aeb0a9661b70e46c23d9f82317754063a5699656227ab28ee04fb3ceb7e36445#action is an example ActionResult executed locally and cached remotely.

Reply:
sorry could you elaborate? I'm not sure what I should be looking for in that invocation link, or how this change is related to local execution

Reply:
I don't think my change for output_paths affected the symlink stuff at all, but I can take a closer look if you two need another pair of eyes. Though, I do remember that the change I made broke a project depending on some `rules_kotlin` targets, so I'd recommend that you test the example projects [here](https://github.com/bazelbuild/rules_kotlin/tree/master/examples) against your remote execution changes.

Reply:
Discussed with Brandon on Slack on this and shared some of the initial findings I got by running the build locally against our remote cache.

@iain-macdonald Note on the Kotlin project. Let me try running it against my change.

Reply:
Tested against both `dagger` and `trivial` in rules_kotlin example and they don't have the same problem as rules_go's go_path so both passed with/without this change applied.